### PR TITLE
Resolve ILLink warnings in System.ComponentModel.Annotations

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #45623

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]

**Comment by eerhardt** [comment]
@ajcvickers - any concerns on this? I plan on merging later today if I don't hear any feedback.

File: src/libraries/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs [hunk 1]
```diff
@@ -185,7 +189,9 @@ public sealed partial class KeyAttribute : System.Attribute
     [System.AttributeUsageAttribute(System.AttributeTargets.Field | System.AttributeTargets.Parameter | System.AttributeTargets.Property, AllowMultiple=false)]
     public partial class MaxLengthAttribute : System.ComponentModel.DataAnnotations.ValidationAttribute
     {
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Uses reflection to get the 'Count' property on types that don't implement ICollection. This 'Count' property may be trimmed.")]
         public MaxLengthAttribute() { }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Uses reflection to get the 'Count' property on types that don't implement ICollection. This 'Count' property may be trimmed.")]
```
Comment (stephentoub) [inline src/libraries/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs:194]: Earlier the message gave advice, e.g. "Ensure it is preserved."  Here we're just stating facts "This Count property may be trimmed".  It'd be nice to be consistent across all such messages.

**Comment by ajcvickers** [comment]
@eerhardt No cocerns.

**Review by joperezr — APPROVED** [review]

File: src/libraries/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs]: Thanks, fixed.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**